IGNITE-14510 Documentation: java thin client continuous queries (#9214)
apache/ignite: docs/_docs/thin-clients/java-thin-client.adoc

Refer to the link:key-value-api/binary-objects[Working with Binary Objects] page for detailed information.

> === Cache entry listening

Reviewer: ```suggestion
=== Cache Entry Listening
```

> When a cache gets modified (an entry is inserted, updated, deleted or expired), an event can be sent to notify the client.
> To listen to these events two approaches can be used:

Reviewer: ```suggestion
When a cache is modified (an entry is inserted, updated, deleted, or expired), an event can be sent to notify the client.
To listen to these events, you can use one of the following approaches:
```

* Continuous queries
* Cache `registerCacheEntryListener` methods

Both approaches require a local listener to be provided, which is triggered on every cache modification event.

For both approaches you can also specify a remote filter to narrow down the range of entries that are monitored for updates. This filter is executed for each updated entry on the server-side and evaluates whether the event should be propagated to the client's local listener.

NOTE: The classes of the remote filter factory must be available on the server nodes of the cluster.

Refer to the link:key-value-api/continuous-queries[thick client continuous queries] page for more information about continuous queries.

> In case of connection to server failure, a thin client can't silently reconnect with guarantees that no events were lost, so continuous queries and registered cache event listeners are closed after the server disconnection. There are methods exist with the additional parameter: disconnect listener. This listener allows to catch server disconnection events and react appropriately.

Reviewer: ```suggestion
In case of connection to server failure, a thin client cannot silently reconnect with guarantees that no events are lost, so continuous queries and registered cache event listeners are closed after the server disconnection. There are also several methods with the additional parameter: disconnect listener. This listener allows to catch server disconnection events and react appropriately.
```
